### ISSUE TITLE
Change the way horizon line is drawned on OSD when banked at 90 degrees to conform with aviation standards

### ISSUE THREAD

**Issue by dxs94** [ArduPilot/ardupilot#25520]
## Feature request

Please note this is different from issue #24855 which was fixed in master.

**Is your feature request related to a problem? Please describe.**
When the aircraft is banked at 90°, the OSD artificial horizon line is not drawn the way it should be according to aviation standards. In this situation (90° of bank), the OSD no longer gives any indication of current pitch attitude. The drawing is the same no matter if the nose is below, on, or above horizon.

**Describe the solution you'd like**
Modify the way OSD horizon line is drawn.

**Platform**
[  ] All
[  ] AntennaTracker
[X] Copter
[X] Plane
[  ] Rover
[  ] Submarine

**Additional context**

When bank is close to 0 horizon line is correct:

![Capture d'écran 2023-11-12 222048](https://github.com/ArduPilot/ardupilot/assets/88461305/dd347b1f-2d0d-4589-ad5a-a2fdeac4adbc)

----------------------------

Now at bank 90°, the OSD line is always the same. No way of knowing if aircraft nose is below, on, or above horizon.

![Capture d'écran 2023-11-12 221942](https://github.com/ArduPilot/ardupilot/assets/88461305/d8dc3e53-49f9-4b49-923a-398dc2ae12ed)

![Capture d'écran 2023-11-12 222443](https://github.com/ArduPilot/ardupilot/assets/88461305/283d7e3a-66af-4f52-8fef-3aed78bd311e)

![Capture d'écran 2023-11-12 223550](https://github.com/ArduPilot/ardupilot/assets/88461305/8bba9a4f-edeb-413e-9124-6688e31ecde6)

----------------------------

When bank is not 0, the horizon line should be drawn sideways to give pitch information. Here is a model of how it should look like:

![OSD roll depiction](https://github.com/ArduPilot/ardupilot/assets/88461305/07af1fa5-71bc-442f-a6de-2025eb170699)

**Comment by Hwurzburg** [comment]
osd/ahrs uses euler angles...pitch becomes undefined at 90 roll....euler angle limitation...roll is undefined at +/-90 deg pitch conversely
either one becomes yaw
this is not a rotating ball like an aviation gyro....

**Comment by dxs94** [comment]
Yes I'm the one who opened the issue regarding pitch which led to PR #5497. When I test flew it I realized there was room for further improvement.

Here I'm considering a further improvement to OSD artificial horizon which could be incorporated at the same time in upcoming AP 4.5. I'm aware of euler limitations, and at 90° bank, pitch is indeed controlled by yaw input. However please consider this: aircraft pitch body attitude is still defined at 90° bank, and this is what should be presented to the pilot (ie. drawn on OSD). Just how it is in mission planner: in below screenshots the pitch value as known to Ardupilot is reported, and the artificial horizon drawing is correct, unlike on OSD.

Level flight, roll 0:
![Demo1](https://github.com/ArduPilot/ardupilot/assets/88461305/469ef210-d26e-44bf-b589-ade187188595)

-----------------
Level flight, roll 90°: the artificial horizon line is in the middle of the attitude display frame:
![Demo2](https://github.com/ArduPilot/ardupilot/assets/88461305/cf7b5d32-cc30-42a8-b087-821eed11447c)

------------------
Aircraft in a climb, right roll 90°: the artificial horizon line is drawn on the RIGHT of the attitude display frame:
![Demo3](https://github.com/ArduPilot/ardupilot/assets/88461305/2678634b-dea1-4d6a-9b1b-4e4fe08fba64)

--------------------
Aircraft in a descent, right roll 90°: the artificial horizon line is drawn on the LEFT of the attitude display frame:
![Demo4](https://github.com/ArduPilot/ardupilot/assets/88461305/7a49f607-86bb-4cb0-8676-f99405971277)

This obviously is reversed for a LEFT 90° roll.

**Comment by Hwurzburg** [comment]
without major code changes 90 deg roll will have gimbal locked pitch, as will 90 deg pitch have gimbal locked roll...not something I would anticipate happening anytime soon, if ever

**Comment by peterbarker** [comment]
> without major code changes 90 deg roll will have gimbal locked pitch, as will 90 deg pitch have gimbal locked roll...not something I would anticipate happening anytime soon, if ever

The AHRS is quite happy to hand you back a quaternion.  I guess the problems lie in the OSD code, then?
